src/hotspot/share/memory/archiveBuilder.hpp

```diff
@@ -161,7 +205,7 @@ class ArchiveBuilder : public StackObj {
   DumpAllocStats* _alloc_stats;
 
   // For global access.
-  static ArchiveBuilder* _singleton;
+  static ArchiveBuilder* _current;
```

Comment:
This is kind of a strange concept since ArchiveBuilder is a StackObj.  Maybe it should just be static if only one thread uses it at a time?

Reply:
And can a different thread call current() ? yikes.

Reply:
ArchiveBuilder has temporarily allocated objects such as `ArchiveBuilder::_ptrmap`. These are easily freed when the ArchiveBuilder falls out of scope. For example, in `DynamicArchive::dump()`.

`current()` should be callable only inside the VM thread, where the dumping operations take place. I'll add an assert for that.